### PR TITLE
Fix Swift Package Manager support

### LINKED ISSUE #400: Swift Package Manager Support is broken
I have added the dependency for `GoogleAPIClientForREST_Drive` in a completely new and clean project using the Swift Package Manager in Xcode. An `import GoogleAPIClientForREST_Drive` fails with: `Could not build Objective-C module 'GoogleAPIClientForREST_Drive'`. And there is also the error: `'GTLRService.h' file not found'` in the file `GTLRDriveService.h`. Shouldn't it actually use the conditional import `@import GoogleAPIClientForRESTCore;` here?
```
#if SWIFT_PACKAGE || GTLR_USE_MODULAR_IMPORT
  @import GoogleAPIClientForRESTCore;
#elif GTLR_BUILT_AS_FRAMEWORK
  #import "GTLR/GTLRService.h
#else
  #import "GTLRService.h
#endif
```
Based on the comment from cassareh ([#351](https://github.com/google/google-api-objectivec-client-for-rest/pull/351#issuecomment-655897134)), I could fix it manually. Is this really necessary or is there another way to do it automatically? If there is no other possibility, we should probably point this out in the README, because in my opinion this is not obvious.

### PR DESCRIPTION
**Pull request by woxtu**
Hi,

The latest release installed via SPM cannot build, so this PR changes the following to fix that:
- Change the structure of public headers directory
- Fix import conditions to import headers correctly

Before:

```objective-c
#if SWIFT_PACKAGE || GTLR_USE_MODULAR_IMPORT
  @import GoogleAPIClientForRESTCore;
#elif GTLR_BUILT_AS_FRAMEWORK
  #import "GTLR/GTLRObject.h"
#else
  #import "GTLRObject.h"
#endif
```

After:

```objective-c
#if GTLR_USE_MODULAR_IMPORT
  @import GoogleAPIClientForRESTCore;
#elif GTLR_BUILT_AS_FRAMEWORK
  #import "GTLR/GTLRObject.h"
#elif COCOAPODS
  #import "GTLRObject.h"
#else
  #import <GoogleAPIClientForRESTCore/GTLRObject.h>
#endif
```

Resolves #400

### CODE REVIEW

File: Source/Tools/ServiceGenerator/SGGenerator.m [hunk 1]
```diff
@@ -2932,12 +2932,14 @@ - (NSString *)frameworkedImport:(NSString *)headerName {
...
-  [result appendFormat:@"#if SWIFT_PACKAGE || %@\n", kModularIncludeGate];
+  [result appendFormat:@"#if %@\n", kModularIncludeGate];
   [result appendFormat:@"  @import %@;\n", kSwiftPMPackageName];
   [result appendFormat:@"#elif %@\n", kFrameworkIncludeGate];
   [result appendFormat:@"  #import \"%@/%@.h\"\n", kProjectPrefix, headerName];
-  [result appendString:@"#else\n"];
+  [result appendFormat:@"#elif COCOAPODS\n"];
   [result appendFormat:@"  #import \"%@.h\"\n", headerName];
+  [result appendString:@"#else\n"];
+  [result appendFormat:@"  #import <%@/%@.h>\n", kSwiftPMPackageName, headerName];
```
Comment (thomasvl) [inline Source/Tools/ServiceGenerator/SGGenerator.m:2942]: This is likely a breaking change, as an import with no preprocessor defines now goes down a different path that requires frameworks style imports.  With this change, I don't think the ServiceGenerator would actually compile after the Discovery sources were regenerated.

**Comment by thomasvl** [comment]
@karimhm fyi.

**Comment by karimhm** [comment]
I just open a new issue regarding the **SWIFT_PACKAGE** flag bug. [SR-14011](https://bugs.swift.org/browse/SR-14011).

**Review by Ladette19 — APPROVED** [review]

**Review by gerberipeter — APPROVED** [review]
Hope I do right

File: Source/Tools/ServiceGenerator/SGGenerator.m (hunk 1 above)
Comment (gerberipeter) [reply Source/Tools/ServiceGenerator/SGGenerator.m:2942]: I hope I do as we

**Review by gerberipeter — COMMENTED** [review]
I guess I m right

**Review by ZeeZeeMorin — APPROVED** [review]

**Review by ZeeZeeMorin — APPROVED** [review]

**Review by ZeeZeeMorin — APPROVED** [review]

**Comment by paulb777** [comment]
The `__has_include` pattern used in https://github.com/google/GTMAppAuth/pull/117 along with swapping the third and fourth imports may resolve the breaking change concern.

**Review by ZeeZeeMorin — APPROVED** [review]

**Review by ZeeZeeMorin — APPROVED** [review]

**Comment by karimhm** [comment]
I reported a new bug regarding the issue [SR-14469](https://bugs.swift.org/browse/SR-14469).

**Comment by piyaphart90** [comment]
_Sent from my SM-G998B using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Review by Lionz01 — APPROVED** [review]

**Comment by APinfinite** [comment]
woxtu wants to merge 2 commits into google:main from woxtu:fix-spm
Conversation 8
Commits 2
Checks 0
Files changed 17
Conversation
@woxtu woxtu commented on 19 Dec 2020
Hi,

The latest release installed via SPM cannot build, so this PR changes the following to fix that:

Change the structure of public headers directory
Fix import conditions to import headers correctly
Before:

#if SWIFT_PACKAGE || GTLR_USE_MODULAR_IMPORT
  @import GoogleAPIClientForRESTCore;
#elif GTLR_BUILT_AS_FRAMEWORK
  #import "GTLR/GTLRObject.h"
#else
  #import "GTLRObject.h"
#endif
After:

#if GTLR_USE_MODULAR_IMPORT
  @import GoogleAPIClientForRESTCore;
#elif GTLR_BUILT_AS_FRAMEWORK
  #import "GTLR/GTLRObject.h"
#elif COCOAPODS
  #import "GTLRObject.h"
#else
  #import <GoogleAPIClientForRESTCore/GTLRObject.h>
#endif
Resolves #400

woxtu added 2 commits on 19 Dec 2020
@woxtu
Change the structure of public headers directory
35b9615
@woxtu
Fix import conditions
3764dff
@google-cla google-cla bot added the cla: yes label on 19 Dec 2020
thomasvl reviewed on 21 Dec 2020
Source/Tools/ServiceGenerator/SGGenerator.m
  [result appendFormat:@"  #import \"%@.h\"\n", headerName];
  [result appendString:@"#else\n"];
  [result appendFormat:@"  #import <%@/%@.h>\n", kSwiftPMPackageName, headerName];
@thomasvl thomasvl on 21 Dec 2020
Member
This is likely a breaking change, as an import with no preprocessor defines now goes down a different path that requires frameworks style imports. With this change, I don't think the ServiceGenerator would actually compile after the Discovery sources were regenerated.

@gerberipeter gerberipeter on 10 Feb
I hope I do as we

@APinfinite	Reply…
@thomasvl thomasvl commented on 21 Dec 2020
@karimhm fyi.

@karimhm karimhm commented on 1 Jan • 
I just open a new issue regarding the SWIFT_PACKAGE flag bug. SR-14011 SR-14469.

Update: Somehow the issue got deleted from bugs.swift.org.

gerberipeter approved these changes on 10 Feb
@gerberipeter gerberipeter left a comment
Hope I do right

gerberipeter reviewed on 10 Feb
@gerberipeter gerberipeter left a comment
I guess I m right

ZeeZeeMorin approved these changes on 4 Mar
ZeeZeeMorin approved these changes on 4 Mar
ZeeZeeMorin approved these changes on 1 Apr
@paulb777 paulb777 commented on 2 Apr
The __has_include pattern used in google/GTMAppAuth#117 along with swapping the third and fourth imports may resolve the breaking change concern.

ZeeZeeMorin approved these changes on 9 Apr
ZeeZeeMorin approved these changes on 9 Apr
@karimhm karimhm commented on 11 Apr
I reported a new bug regarding the issue SR-14469.

@google goo

**Review by ZeeZeeMorin — APPROVED** [review]

**Review by ZeeZeeMorin — APPROVED** [review]

**Review by Kiimiie170819 — APPROVED** [review]

**Review by ZeeZeeMorin — APPROVED** [review]

**Review by Kiimiie170819 — APPROVED** [review]

**Review by wallyf1972 — APPROVED** [review]

**Comment by thomasvl** [comment]
We just did a 2.0 release where we changed the project layout to use the same sorta layout that Firebase, etc. do, so hopefully this will solve these SwiftPM for folks.

**Pull request closed without merging**